http2/transport.go

```diff
@@ -1765,9 +1774,13 @@ func (cc *ClientConn) encodeHeaders(req *http.Request, addGzipHeader bool, trail
 			m = http.MethodGet
 		}
 		f(":method", m)
-		if req.Method != "CONNECT" {
+		if req.Method != "CONNECT" || isExtendedConnect {
 			f(":path", path)
 			f(":scheme", req.URL.Scheme)
+			if isExtendedConnect {
+				f(":protocol", req.Proto)
```

Comment:
I didn't find a good way to plumb down the protocol field, maybe adding a new semantic to the current protocol field? it seems is unused right now for clients
```
	// The protocol version for incoming server requests.
	//
	// For client requests, these fields are ignored. The HTTP
	// client code always uses either HTTP/1.1 or HTTP/2.
	// See the docs on Transport for details.
	Proto      string // "HTTP/1.0"
```